diff --git a/library.mak b/library.mak
--- a/library.mak
+++ b/library.mak
@@ -13,10 +13,16 @@ all-$(CONFIG_SHARED): $(SUBDIR)$(SLIBNAME)
 LIBOBJS := $(OBJS) $(SUBDIR)%.h.o $(TESTOBJS)
 $(LIBOBJS) $(LIBOBJS:.o=.s) $(LIBOBJS:.o=.i):   CPPFLAGS += -DHAVE_AV_CONFIG_H
 $(TESTOBJS) $(TESTOBJS:.o=.i): CFLAGS += -Umain

 $(SUBDIR)$(LIBNAME): $(OBJS)
-	$(RM) $@
-	$(AR) $(ARFLAGS) $(AR_O) $^
-	$(RANLIB) $@
+	$(RM) $@
+ifneq ($(findstring emar,$(AR)),)
+	$(file >$@.rsp,$^)
+	$(AR) $(ARFLAGS) $(AR_O) @$@.rsp
+	$(RM) $@.rsp
+else
+	$(AR) $(ARFLAGS) $(AR_O) $^
+endif
+	$(RANLIB) $@

 install-headers: install-lib$(NAME)-headers install-lib$(NAME)-pkgconfig
